--- a/makeicecat
+++ b/makeicecat
@@ -31,9 +31,12 @@
 
 DATA="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"/data
 
-mkdir -p output
+[ -d output ] || [ "$UNPACK" ] || export UNPACK=1
+[ -z "$UNPACK" ] || mkdir -p output
 cd output
 
+if [ "$UNPACK" ]; then
+
 ###############################################################################
 # Retrieve FF source code
 ###############################################################################
@@ -54,6 +57,8 @@
 # Retrieve l10n
 ###############################################################################

+[ ! -f "$DATA/shipped-locales" ] ||
+cp "$DATA/shipped-locales" "$SOURCEDIR/browser/locales/shipped-locales"
 mkdir l10n
 cd l10n
 while read line;do
@@ -76,6 +81,10 @@
 rm compare-locales/.hg* compare-locales/.git* -rf
 mv compare-locales $SOURCEDIR/l10n
 
+if [ "$UNPACK" = 'only' ]; then exit 0; fi
+
+fi
+
 #######################################################
 
 cd $SOURCEDIR
