--- a/makeicecat
+++ b/makeicecat
@@ -29,9 +29,12 @@

 DATA="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"/data

-mkdir output
+[ -d output ] || [ "$UNPACK" ] || export UNPACK=1
+[ -z "$UNPACK" ] || mkdir output
 cd output

+if [ "$UNPACK" ]; then
+
 ###############################################################################
 # Retrieve FF source code
 ###############################################################################
@@ -75,6 +78,10 @@
 rm compare-locales/.hg* compare-locales/.git* -rf
 mv compare-locales $SOURCEDIR/l10n

+if [ "$UNPACK" = 'only' ]; then exit 0; fi
+
+fi
+
 #######################################################

 cd $SOURCEDIR
